### ISSUE TITLE
Kind of connect with the youtube app but the sound is beeing played on the phone

### ISSUE THREAD

**Issue by basilelt** [patrickkfkan/volumio-ytcr#1]
Everything seems to work fine except when I try to play music. Here is a screenshot of journalctl, no more informations given.
![image](https://user-images.githubusercontent.com/66386196/131229031-071fbe74-3905-4a1e-aba9-bcc7d661f948.png)

**Comment by wother** [comment]
Hey friend! Same issue.

When hopping into debug the following is thrown:

`(node:831) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'name' of null
    at EventEmitter. (/data/plugins/music_service/ytcr/index.js:237:76)
    at EventEmitter.emit (events.js:315:20)
    at EventEmitter. (/data/plugins/music_service/ytcr/node_modules/yt-cast-receiver/lib/index.js:52:16)
    at EventEmitter.emit (events.js:315:20)
    at YouTubeApp.stop (/data/plugins/music_service/ytcr/node_modules/yt-cast-receiver/lib/app.js:158:31)
    at runMicrotasks ()
    at processTicksAndRejections (internal/process/task_queues.js:93:5)`

This is from the network interfaces portion of index.js. That dependency may have an issue with something up the pipe.

**Comment by patrickkfkan** [comment]
I just tried casting from YouTube app to this plugin and it does so without issue. This makes it difficult for me to debug, other than suggesting you restart Volumio, your phone and router...

**Comment by wother** [comment]
I did some digging, the dev console is SUPER helpful here. My issue was that the YT app cast perfectly (so the plugin does indeed do what it says on the tin) but the YT Music app casts differently. I was hoping for the ability to cast from YT music, as it was tangentially related, but alas it seems that I will need to do some actual work to get it going.

If I can get it working within the bounds of the current plugin, without hurting the abbility to cast from the YT app, I will put in a pull request with my changes. If I end up getting an "either or" situation, I will probably fork. (within bounds of license and distribution rights, of course)

TLDR: Plugin works from YT app, but not YT music app. This issue should likely be closed, unless OP has more information about it.

**Comment by patrickkfkan** [comment]
YT uses CAST v1 protocol which is still supported probably due to the number of legacy "smart" TVs out there.

YT Music as I understand uses a later version of the CAST protocol, and performs device authentication against the cast receiver. A key from Google is required for authentication to succeed and this key is only available when you register your device / app with Google. I don't think us "laymen" are eligible to apply.

I would love to be proved wrong. Having the ability to cast YT Music to Volumio would be a great feature indeed.

**Comment by Sperryfreak01** [comment]
I've been trying to find a path forward for this because its all I want for Christmas.  I'm not sure if this is helpful at all but I see there is a signup page to register a Chromecast receivers.  Not sure if this would get us past the v2 auth.  I can buy a dev account though if it helps, only $5usd. 


https://cast.google.com/publish/#/signup

**Comment by LukaHummel** [comment]
> I've been trying to find a path forward for this because its all I want for Christmas.  I'm not sure if this is helpful at all but I see there is a signup page to register a Chromecast receivers.  Not sure if this would get us past the v2 auth.  I can buy a dev account though if it helps, only $5usd. 
> 
> 
> https://cast.google.com/publish/#/signup

Did you try to register your volumio device?
